fe/fe-core/src/main/java/com/starrocks/sql/common/PRangeCell.java

```diff
@@ -63,29 +57,32 @@ public int compareTo(PartitionRange o) {
      *         && other.lowerBound.compareTo(upperBound) <= 0;
      *   }
      */
-    public boolean isIntersected(PartitionRange o) {
-        return this.partitionKeyRange.upperEndpoint().compareTo(o.partitionKeyRange.lowerEndpoint()) > 0 &&
-                this.partitionKeyRange.lowerEndpoint().compareTo(o.partitionKeyRange.upperEndpoint()) < 0;
+    public boolean isIntersected(PRangeCell o) {
+        return this.range.upperEndpoint().compareTo(o.range.lowerEndpoint()) > 0 &&
```

Comment:
It's a bit different
```
    /**
     * Check two partition range is `interact` which is a bit different from Range's `isConnected` method, eg:
     * [2, 4) and [4, 6) are not interact;
     * [2, 4) and [4, 6) are connected, because both enclose the empty range [4, 4).
     *
     * public boolean isConnected(Range<C> other) {
     *     return lowerBound.compareTo(other.upperBound) <= 0
     *         && other.lowerBound.compareTo(upperBound) <= 0;
     *   }
     */
```